File: src/main/java/kr/syeyoung/dungeonsguide/dungeon/roomprocessor/RoomProcessor.java
package kr.syeyoung.dungeonsguide.dungeon.roomprocessor;

import kr.syeyoung.dungeonsguide.events.impl.BlockUpdateEvent;
import kr.syeyoung.dungeonsguide.events.impl.KeyBindPressedEvent;
import kr.syeyoung.dungeonsguide.events.impl.PlayerInteractEntityEvent;
import net.minecraft.util.IChatComponent;
import net.minecraftforge.client.event.GuiScreenEvent;
import net.minecraftforge.event.entity.living.LivingDeathEvent;
import net.minecraftforge.event.entity.living.LivingEvent;
import net.minecraftforge.event.entity.player.PlayerInteractEvent;

public interface RoomProcessor {
    void tick();

    void drawScreen(float partialTicks);

    void drawWorld(float partialTicks);

    void chatReceived(IChatComponent chat);

    void actionbarReceived(IChatComponent chat);

    boolean readGlobalChat();

    void onPostGuiRender(GuiScreenEvent.DrawScreenEvent.Post event);

    void onEntityUpdate(LivingEvent.LivingUpdateEvent updateEvent);

    void onEntityDeath(LivingDeathEvent deathEvent);

    void onKeybindPress(KeyBindPressedEvent keyInputEvent);

    void onInteract(PlayerInteractEntityEvent event);

    void onInteractBlock(PlayerInteractEvent event);

    void onBlockUpdate(BlockUpdateEvent blockUpdateEvent);
}

File: src/main/java/kr/syeyoung/dungeonsguide/dungeon/roomprocessor/RoomProcessor.kt
package kr.syeyoung.dungeonsguide.dungeon.roomprocessor

import kr.syeyoung.dungeonsguide.events.impl.BlockUpdateEvent
import kr.syeyoung.dungeonsguide.events.impl.KeyBindPressedEvent
import kr.syeyoung.dungeonsguide.events.impl.PlayerInteractEntityEvent
import net.minecraft.util.IChatComponent
import net.minecraftforge.client.event.GuiScreenEvent.DrawScreenEvent
import net.minecraftforge.event.entity.living.LivingDeathEvent
import net.minecraftforge.event.entity.living.LivingEvent.LivingUpdateEvent
import net.minecraftforge.event.entity.player.PlayerInteractEvent

interface RoomProcessor {
    fun tick()
    fun drawScreen(partialTicks: Float)
    fun drawWorld(partialTicks: Float)
    fun chatReceived(chat: IChatComponent)
    fun actionbarReceived(chat: IChatComponent)
    fun readGlobalChat(): Boolean
    fun onPostGuiRender(event: DrawScreenEvent.Post)
    fun onEntityUpdate(updateEvent: LivingUpdateEvent)
    fun onEntityDeath(deathEvent: LivingDeathEvent)
    fun onKeybindPress(keyInputEvent: KeyBindPressedEvent)
    fun onInteract(event: PlayerInteractEntityEvent)
    fun onInteractBlock(event: PlayerInteractEvent)
    fun onBlockUpdate(blockUpdateEvent: BlockUpdateEvent)
}